webui/react/src/utils/types.ts

```diff
@@ -281,10 +281,13 @@ export const getMetricValue = (workload?: Workload, metricName?: string): number
 };
 
 export const getBatchNumber = (
-  data: {batch: number} | {numBatches: number, priorBatchesProcessed: number},
+  data: {batch: number} | {numBatches: number, priorBatchesProcessed: number}
+  | {totalBatches: number},
 ): number => {
   if ('batch' in data) {
     return data.batch;
+  } else if ('totalBatches' in data) {
+    return data.totalBatches;
```

Comment:
**suggestion:** I would just get rid of this `getBatchNumber` function entirely and change the only file that references this function `src/components/CheckpointModal.tsx`
* remove `getBatchNumber` from line 11 (remove the import from `utils/types`)
* change line 87 to be `const totalBatchesProcessed = checkpoint.totalBatches;`

Reply:
@hkang1 the input checkpoint for that CheckpointModal is `CheckpointWorkloadExtended | CheckpointDetail`, of which `CheckpointDetail` doesn't have totalBatches, so I think that was the point? I'll just leave this to you if that's ok..

Reply:
ah yeah, you're right. how you have it is great 👍